packages/protocol/contracts/L1/ProofVerifier.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+import {AddressResolver} from "../common/AddressResolver.sol";
+import {Proxied} from "../common/Proxied.sol";
+import {TaikoErrors} from "./TaikoErrors.sol";
+import {TaikoData} from "./TaikoData.sol";
+import {LibVerifyTrusted} from "./libs/proofTypes/LibVerifyTrusted.sol";
+import {LibVerifyZKP} from "./libs/proofTypes/LibVerifyZKP.sol";
+
+library TaikoProofToggleMask {
+    function getToggleMask() internal pure returns (uint16) {
+        // BITMAP for efficient iteration and flexible additions later
+        // ZKP_ONLY,            // 0000 0001
+        // SGX_ONLY,            // 0000 0010
+        // RESERVED_X_ONLY,     // 0000 0100
+        // RESERVED_Y_ONLY,     // 0000 1000
+        // ZKP_AND_SGX,         // 0000 0011
+        // X_ZKP_SGX,           // 0000 0111
+        return uint16(1); // ZKP ONLY by default
+    }
+}
+
+/// @custom:security-contact hello@taiko.xyz
+contract ProofVerifier is TaikoErrors {
+    uint256[50] private __gap;
+
+    function getToggleMask() public pure virtual returns (uint16) {
+        return TaikoProofToggleMask.getToggleMask();
+    }
+
+    function verifyProofs(
+        bytes32 instance, 
+        TaikoData.TypedProof[] calldata blockProofs,
+        AddressResolver resolver
+    )
+    external
+    view
+    {
+        uint16 mask = getToggleMask();
```

Comment:
Maybe we should simply remove the mask? A verifier just always hard-codes the logic that needs to be followed. If this logic changes, a different verifier contract should be deployed.

Well, it is maybe useful for testing still, so this could be the testnet verifier contract or something like that. But on mainnet, probably just want to keep it as simple as possible.